src/legacy/core_plugins/kibana/public/dashboard/panel/dashboard_panel.test.tsx

```diff
@@ -17,38 +17,40 @@
  * under the License.
  */
 
-import React from 'react';
+// TODO: remove this when EUI supports types for this.
+// @ts-ignore: implicit any for JS file
+import { takeMountedSnapshot } from '@elastic/eui/lib/test';
 import _ from 'lodash';
+import React from 'react';
+import { Provider } from 'react-redux';
 import { mountWithIntl } from 'test_utils/enzyme_helpers';
-import { DashboardPanel } from './dashboard_panel';
-import { DashboardViewMode } from '../dashboard_view_mode';
-import { PanelError } from '../panel/panel_error';
 import { store } from '../../store';
+// @ts-ignore: implicit any for JS file
 import { getEmbeddableFactoryMock } from '../__tests__/get_embeddable_factories_mock';
-
 import {
-  updateViewMode,
+  embeddableError,
+  embeddableIsInitialized,
+  embeddableIsInitializing,
+  embeddableStateChanged,
   setPanels,
   updateTimeRange,
-  embeddableIsInitialized,
+  updateViewMode,
 } from '../actions';
-import { Provider } from 'react-redux';
+import { DashboardViewMode } from '../dashboard_view_mode';
+import { DashboardPanel, DashboardPanelUiProps } from './dashboard_panel';
 
-import {
-  takeMountedSnapshot,
-} from '@elastic/eui/lib/test';
+import { PanelError } from './panel_error';
 
-function getProps(props = {}) {
+function getProps(props = {}): DashboardPanelUiProps {
   const defaultTestProps = {
     panel: { panelIndex: 'foo1' },
     viewOnlyMode: false,
-    destroy: () => {},
     initialized: true,
     lastReloadRequestTime: 0,
-    embeddableIsInitialized: () => {},
-    embeddableIsInitializing: () => {},
-    embeddableStateChanged: () => {},
-    embeddableError: () => {},
+    embeddableIsInitialized,
+    embeddableIsInitializing,
+    embeddableStateChanged,
+    embeddableError,
```

Comment:
If you're going to be using the defined actions, I think you don't need to list them here.  I think these are default props that override what would normally get passed in from state (hence why the original implementation lists these functions with no actual functionality).  (I'm not sure what's need for the test to run correctly, though, so this is also a question on my part about what's necessary :) )

Reply:
This worked great in terms of the tests running locally - so I agree that it makes sense to cut these